packages/@dcl/ecs/src/runtime/initialization/dcl.ts

```diff
@@ -0,0 +1,47 @@
+/// <reference types="@dcl/posix" />
+
+import { IEngine } from '../../engine'
+import { _initEventObservables } from '../observables'
+import { Transport } from '../types'
+
+export function initializeDcl(engine: IEngine, rendererTransport: Transport) {
+  if (typeof dcl !== 'undefined') {
+    dcl.loadModule('~system/ExperimentalApi', {}).catch(dcl.error)
+
+    async function pullRendererMessages() {
+      const response = await dcl.callRpc(
+        '~system/ExperimentalApi',
+        'messageFromRenderer',
+        []
+      )
+
+      if (response.data?.length) {
+        if (rendererTransport.onmessage) {
+          for (const byteArray of response.data) {
+            rendererTransport.onmessage(byteArray)
+          }
+        }
+      }
+    }
+
+    dcl.onUpdate((dt: number) => {
```

Comment:
onUpdate should now accept an async function and should block the next update until the promise is completed

Reply:
pullRendererMessages should be awaited, but tasks can be deferred through the event loop and not in the system's pipeline